frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-page.tsx

```diff
@@ -90,33 +97,40 @@ const InstallCluster: React.FC<InstallClusterProps> = ({ match }) => {
     return sanitizedMode;
   };
 
-  const getStep = (offset: number = 0) => {
-    const searchParams = new URLSearchParams(window.location.search.slice(1));
-    const step = parseInt(searchParams.get('step'), 10) || 1;
-    const sanitizedStep = step && step <= 5 - offset && step >= 1 ? step : 1;
-    return sanitizedStep;
-  };
-
   const getParamString = (step: number, mode: number) => {
     const searchParams = new URLSearchParams(window.location.search.slice(1));
     searchParams.set('step', step.toString());
     searchParams.set('mode', mode.toString());
     return searchParams.toString();
   };
 
+  const getAnchor = (step: number, mode: number) => `~new?${getParamString(step, mode)}`;
+
+  const getStep = (offset: number = 0) => {
+    const searchParams = new URLSearchParams(window.location.search.slice(1));
+    const step = parseInt(searchParams.get('step'), 10) || 1;
+    const sanitizedStep = step && step <= 5 - offset && step >= 1 ? step : 1;
+    return sanitizedStep;
+  };
+
   const getIndex = (searchSpace: any, search: string, offset: number = 0) => {
     const index = Object.values(searchSpace).findIndex((el) => el === search);
     return index - offset + 1;
   };
 
-  const getAnchor = (step: number, mode: number) => `~new?${getParamString(step, mode)}`;
-
   const handleModeChange = (event: React.FormEvent<HTMLInputElement>) => {
     const { value } = event.currentTarget;
     const modeIndex = getIndex(MODES, value);
-    history.push(
-      `~new?${getParamString(getIndex(CreateStepsSC, CreateStepsSC.DISCOVER), modeIndex)}`,
-    );
+    switch (modeIndex) {
+      case 2:
+        if (hasNoProvSC) {
+          history.push(getAnchor(3, 2));
+          break;
+        }
```

Comment:
```suggestion
        if (hasNoProvSC) history.push(getAnchor(3, 2));
        break;
```

Reply:
In cases where `modeIndex` is 2 and `hasNoProvSC` is false, the control flow will encounter `break` and exit thus not falling back to the default condition, hence the basic functionality will be affected, no? Also, seeing that we need to fallback to a default condition, using `if`s here is a better choice since we can have as many cases as we want that fallback to a default condition. Making changes.